compiler-core/src/type_/environment.rs

```diff
@@ -55,6 +56,7 @@ pub enum EntityKind {
     ImportedType,
     ImportedTypeAndConstructor,
     ImportedValue,
+    ImportedModule,
```

Comment:
The usage tracking system is fairly complex because it handles scopes, but modules can't be imported in a scoped fashion so we don't need to use it here. Instead we can store a set of imported modules in the typing environment can delete them from there as they are seen. This will be simpler and won't result in false negatives from treating the two namespaces as one.

Later this scoped usage tracking system will be deleted and replaced with a GC style graph traversal which can detect cycles, so I'd prefer not to put more items in there.

Reply:
@lpil Sounds good. I'll update to use that technique instead